ansible_collections/arista/avd/plugins/action/anta_runner.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+
+from ansible.plugins.action import ActionBase
+from anta.inventory.models import InventoryDevice
+from anta.result_manager import ResultManager
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        device_params = {
+            "name": task_vars["inventory_hostname"],
+            "host": task_vars["ansible_host"],
+            "username": task_vars["ansible_user"],
+            "password": task_vars["ansible_httpapi_password"],
```

Comment:
I think you want to take these from the _connection object and not the input vars, since they need to be de-vaulted etc.